packages/relay/src/lib/eth.ts

```diff
@@ -149,6 +151,7 @@ export class EthImpl implements Eth {
     mirrorNodeClient: MirrorNodeClient,
     logger: Logger,
     chain: string,
+    register: Registry,
     cache?
   ) {
     this.sdkClient = nodeClient;
```

Comment:
I suggest _renaming_ this field.  That will ensure (I think? this is TS not JS so yes?) that _all_ uses of the sdk client are now going through `getSdkClient()` instead of accessing it directly.  Or not?

Reply:
Is @david-bakin-sl comment more so instead of renaming a variable should we create a new instances so that existing the function removes it from use and allows it to be garbage collected?

Reply:
I was reacting to [line 185](https://github.com/hashgraph/hedera-json-rpc-relay/blob/40b1bcc13d15a907f6bb7372e2fba470ffd930c7/packages/relay/src/lib/eth.ts#L185) where it looks like there's an instance variable holding the current SDK client and whenever it times out (call limit exceeded) you create a new sdk client and stomp over the old one.  But of course the old one still exists as long as anyone references it.  So calls _in flight_ still work with the old client they they're holding.  (I guess that's how the framework works, don't know - in other languages mostly a webservice call being handled has local state that disappears only when the webservice call is completed, and _outgoing_ clients would be held in that local state.)  _New_ calls coming in get the new client.  There might be a race condition where two calls come in at nearly the same time and so two new clients are created - one will win (be stored into sdkClient last) and be used for future calls, the other one will only be used for one call then thrown away.  This probably doesn't matter at all.

The point is: I think this code is fine as it is.

And to the injected dependency thing - well, have it inject to the _renamed_ variable's name, right?  Alternatively - just rename it _long enough to do a compile_ and check that it isn't used directly anywhere and let us know you confirmed that.

Reply:
Thank you, good idea to rename it so the reviewer is confident there is not used anywhere where he does not see it explicitly on the changes.

I have rename  it and thanks for the insight, very interesting about the way the references work.